### ISSUE TITLE
[BUG]: The `autocomplete` attribute has an incorrect value for `tel-national`

### ISSUE THREAD

**Issue by alexnozer** [IBMa/equal-access#2228]
### Project

extension-checker

### Browser

Chrome

### Operating system

Windows

### Description

Hello! When checking the form page using the Accessibility Checker extension, I got the violation: "The ‘autocomplete’ attribute has an incorrect value" for the field with `autocomplete="tel-national"` attribute. `tel-national` value [described](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-tel-national) in WHATWG standard and this is valid value. In fact, using it should not cause an violation.

![Image](https://github.com/user-attachments/assets/fa3e43f9-7452-48e2-95cc-42ca9aae1979)

![Image](https://github.com/user-attachments/assets/f88f2f77-f074-42e3-839d-a24b1643c234)

### Steps to reproduce

1. Open [codepen demo](https://cdpn.io/alexnozer/fullpage/mydqPov?nocache=true&view=fullpage)
2. Run Accessibility Checker scan
3. Check Violations

**Comment by tombrunet** [comment]
Codepen testcase looks good. Looks like we have `tel-national` in the tokens list:
https://github.com/IBMa/equal-access/blob/3e7c48eb8d57090583452e0f4b2b37558f8fe3fc/accessibility-checker-engine/src/v4/rules/input_autocomplete_valid.ts#L176

This will need a small hunt since it's not just that the token's missing from our lists.

**Comment by alexnozer** [comment]
> This will need a small hunt since it's not just that the token's missing from our lists.

I checked the code of [this test](https://github.com/IBMa/equal-access/blob/3e7c48eb8d57090583452e0f4b2b37558f8fe3fc/accessibility-checker-engine/src/v4/rules/input_autocomplete_valid.ts) and found that there is an additional check related to the value of the input element type. When the `type="tel"` attribute is set, only tokens from the `tokensMandatoryGroup2_tel` group, which contains only one `tel` token, are allowed.

Also checking the codepen example with [HTML Validator](https://validator.w3.org/nu/) gives the same error:

![Image](https://github.com/user-attachments/assets/e19418fe-0b3b-48d4-bf58-3171ddd12909)

Maybe I'm missing something and the value `type="tel"` somehow affects autocomplete behavior and conflicts with `autocomplete="tel-national"` and therefore counts as an error. This might make sense, but I haven't found any information about it yet. But the presence of check in Accessibility Checker and HTML validator shows that it might not be a bug.

**Comment by alexnozer** [comment]
Meanwhile, I found this in the HTML spec:

> Some fields are only appropriate for certain form controls. An [autofill field](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-field) name is inappropriate for a control if the control does not belong to the group listed for that [autofill field](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-field) in the fifth column of the first row describing that [autofill field](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-field) in the table below. What controls fall into each group is described below the table.

A check of the token table shows that the `tel` token is valid for the `tel` type, while `tel-national` and other sub-values are only valid for the `text` type. So the violation is fair and is not a bug. Therefore, the issue can be closed.
